RecoEgamma/ElectronIdentification/src/ElectronMVAEstimatorRun2Fall17.cc

```diff
@@ -0,0 +1,419 @@
+                                  eleeopout,                // 15
+                                  oneOverEminusOneOverP,
+                                  // Geometrical matchings
+                                  deta,                     // 17
+                                  dphi,
+                                  detacalo,
+                                  // Isolation variables
+                                  pfPhotonIso,
+                                  pfChargedHadIso,
+                                  pfNeutralHadIso,
+                                  // Pileup
+                                  (float)*rho,
+
+                                  // Endcap only variables NOTE: we don't need
+                                  // to check if we are actually in the endcap
+                                  // or not, as it is the last variable in the
+                                  // vector and it will be ignored by the
+                                  // GBRForest for the barrel.
+                                  //
+                                  // The GBRForest classification just needs an
+                                  // array with the input variables in the
+                                  // right order, what comes after doesn't
+                                  // matter.
+                                  preShowerOverRaw          // 24
+                              );
+
+    constrainMVAVariables(vars);
+
+    return vars;
+  }
+  else
+  {
+    std::vector<float> vars = packMVAVariables(
+                                  see,                      // 0
+                                  spp,                      // 1
+                                  circularity,
+                                  r9,
+                                  etawidth,
+                                  phiwidth,                 // 5
+                                  hoe,
+                                  kfhits,
+                                  kfchi2,
+                                  gsfchi2,                  // 9
+                                  fbrem,
+                                  gsfhits,
+                                  expectedMissingInnerHits,
+                                  convVtxFitProbability,     // 13
+                                  eop,
+                                  eleeopout,                // 15
+                                  oneOverEminusOneOverP,
+                                  deta,                     // 17
+                                  dphi,
+                                  detacalo,
+                                  (float)*rho,
+                                  preShowerOverRaw          // 21
+                              );
+
+    constrainMVAVariables(vars);
+
+    return vars;
+  }
+}
+
+void ElectronMVAEstimatorRun2Fall17::constrainMVAVariables(std::vector<float>& vars) const {
+
+  // Check that variables do not have crazy values
```

Comment:
All the constraints here are hardcoded: couldn't you more conveniently make them configurable?